core/src/components/cat-form-group/cat-form-group.tsx

```diff
@@ -0,0 +1,62 @@
+import { Component, Element, h, Host, Prop, Watch } from '@stencil/core';
+
+type HTMLCatFormFieldElement =
+  | HTMLCatInputElement
+  | HTMLCatTextareaElement
+  | HTMLCatSelectElement
+  | HTMLCatLabelElement;
+@Component({
+  tag: 'cat-form-group',
+  styleUrl: 'cat-form-group.scss',
+  shadow: true
+})
+export class CatFormGroup {
+  private formElements?: HTMLCatFormFieldElement[];
+
+  @Element() hostElement!: HTMLElement;
+
+  /**
+   * Whether the labels need a marker to shown if the forms fields are required or optional.<br />
+   *
+   * By default, it is set to auto, it will display the mark depending on the number of required and optional fields: <br />
+   *
+   * If there are more required, the optional will be marked.
+   * If there are less required, it will mark the required.
+   */
+  @Prop() requiredMarker: 'none' | 'required' | 'optional' | 'auto' = 'auto';
+
+  @Watch('requiredMarker')
+  onRequiredMarker(newRequiredMarker: 'none' | 'required' | 'optional' | 'auto') {
+    let updateMarker;
+    if (this.formElements) {
+      updateMarker = newRequiredMarker === 'auto' ? this.calculate(this.formElements) : newRequiredMarker;
+      for (const element of this.formElements) {
+        if (!element.requiredMarker) {
+          element.requiredMarker = updateMarker;
```

Comment:
will this lead to a problem where the change only works once? because the initial change, the requiredMarker is not set, but then it is set by the parent component. If then the requiredMarker changes in the parent, it will skip this component, because it has already been set. Know what I mean? Don't know a good way how to fix it thoguh

Reply:
hmm yes, I understand, I'll try to think of a solution too

Reply:
At the moment if we put the required marker with a default value in each element, my first solution would not work.

A solution that could work, could be to add more options or a variable in the child elements to mark that it cannot be overridden, and that the parent can always modify that variable, except if it is an option or has that mark. For example, if the `requireMarked` has “!important”

Or that the parent can always overwrite the variable without restrictions, and thus only have a source of truth, in the case that a child wants to overwrite that variable, then the structure is changed, that is, that it is not wrapped by the parent anymore.